src/runtime/audience-action-flow-test.js

```diff
@@ -867,528 +781,12 @@ describes.realWin('AudienceActionIframeFlow', (env) => {
-    eventManagerMock
-      .expects('logEvent')
-      .withExactArgs(
-        {
-          eventType: AnalyticsEvent.EVENT_SURVEY_DATA_TRANSFER_FAILED,
-          eventOriginator: EventOriginator.SWG_CLIENT,
-          isFromUserAction: false,
-          additionalParameters: null,
-          configurationId: null,
-        },
-        undefined,
-        undefined
-      )
-      .once();
-
-    await audienceActionFlow.start();
-
-    const failSurveyDataTransferResponse = new SurveyDataTransferResponse();
-    failSurveyDataTransferResponse.setSuccess(false);
-
-    const activityIframeViewMock = sandbox
-      .mock(audienceActionFlow.activityIframeView_)
-      .expects('execute')
-      .withExactArgs(failSurveyDataTransferResponse)
-      .once();
-
-    const messageCallback = messageMap[TEST_SURVEYDATATRANSFERREQUEST.label()];
-    messageCallback(TEST_SURVEYDATATRANSFERREQUEST);
-
-    await tick(10);
-
-    expect(self.console.warn).to.have.been.calledWithExactly(
-      '[swg.js] Exception in publisher provided logging callback: Error: Test Callback Exception'
-    );
-    activityIframeViewMock.verify();
-    onResultMock.verify();
-  });
-
-  it(`handles a SurveyDataTransferRequest with onResult logging rejection`, async () => {
-    const onResultMock = sandbox
-      .mock()
-      .withExactArgs(TEST_SURVEYONRESULTRESPONSE)
-      .rejects(new Error('Test Callback Exception'))
-      .once();
-
-    const audienceActionFlow = new AudienceActionIframeFlow(runtime, {
-      action: 'TYPE_REWARDED_SURVEY',
-      configurationId: TEST_SURVEYONRESULTCONFIGID,
-      onCancel: onCancelSpy,
-      autoPromptType: AutoPromptType.CONTRIBUTION,
-      onResult: onResultMock,
-      calledManually: false,
-    });
-
-    activitiesMock.expects('openIframe').resolves(port);
-
-    eventManagerMock
-      .expects('logEvent')
-      .withExactArgs(
-        {
-          eventType: AnalyticsEvent.EVENT_SURVEY_DATA_TRANSFER_FAILED,
-          eventOriginator: EventOriginator.SWG_CLIENT,
-          isFromUserAction: false,
-          additionalParameters: null,
-          configurationId: null,
-        },
-        undefined,
-        undefined
-      )
-      .once();
-
-    await audienceActionFlow.start();
-
-    const failSurveyDataTransferResponse = new SurveyDataTransferResponse();
-    failSurveyDataTransferResponse.setSuccess(false);
-
-    const activityIframeViewMock = sandbox
-      .mock(audienceActionFlow.activityIframeView_)
-      .expects('execute')
-      .withExactArgs(failSurveyDataTransferResponse)
-      .once();
-
-    const messageCallback = messageMap[TEST_SURVEYDATATRANSFERREQUEST.label()];
-    messageCallback(TEST_SURVEYDATATRANSFERREQUEST);
-
-    await tick(10);
-
-    expect(self.console.warn).to.have.been.calledWithExactly(
-      '[swg.js] Exception in publisher provided logging callback: Error: Test Callback Exception'
+  it('handleSurveyDataTransferRequest called on SurveyDataTransferRequest', async () => {
```

Comment:
If you're going to mock the handler can we remove `storageMock`?

Reply:
Cannot remove storageMock in the tests here since storage is still being accessed in audience-action-flow through `this.deps_.storage()`